core/src/main/scala/akka/persistence/r2dbc/PayloadCodec.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.r2dbc
+
+import java.nio.ByteBuffer
+import io.r2dbc.postgresql.codec.Json
+import io.r2dbc.spi.Row
+import io.r2dbc.spi.Statement
+
+sealed trait PayloadCodec {
+  def encode(bytes: Array[Byte]): AnyRef
+  def decode(payload: AnyRef): Array[Byte]
+}
+object PayloadCodec {
+  case object ByteArrayCodec extends PayloadCodec {
+    override def encode(bytes: Array[Byte]): Array[Byte] = bytes
+    override def decode(payload: AnyRef): Array[Byte] = payload.asInstanceOf[ByteBuffer].array()
```

Comment:
previously this was
```
row.get(name, classOf[Array[Byte]])
```
but now ByteBuffer via
```
row.get(name)
```

I think I would prefer to keep the explicit byte array and let the driver retrieve that in the way it decides is most efficient.

Reply:
Fixed